paramak/parametric_reactors/ball_reactor.py

```diff
@@ -152,6 +165,39 @@ def __init__(
             self.plasma_gap_vertical_thickness,
             self.major_radius - self.minor_radius]
 
+        self.port_type = port_type
+        self.number_of_ports = number_of_ports
+        self.port_center_point = port_center_point
+        self.port_radius = port_radius
+        self.port_height = port_height
+        self.port_width = port_width
+        self.port_distance = port_distance
+        self.port_start_radius = port_start_radius
+        self.port_azimuth_placement_angle = port_azimuth_placement_angle
+        self.port_fillet_radius = port_fillet_radius
+
+        if self.number_of_ports is not None:
+            if self.port_azimuth_placement_angle is None:
+                self.port_azimuth_placement_angle = np.linspace(
```

Comment:
numpy isn't imported here, this should fail

Reply:
Ah yes, good spot! Will add to branch which will be merged into this branch.

Reply:
It's weird that this wasn't caught by the tests